internal/repository/repository_test.go

```diff
@@ -91,8 +91,9 @@ func TestSaveFrom(t *testing.T) {
 }
 
 func BenchmarkSaveAndEncrypt(t *testing.B) {
-	repo, cleanup := repository.TestRepository(t)
+	r, cleanup := repository.TestRepository(t)
 	defer cleanup()
+	repo := r.(*repository.Repository)
```

Comment:
I'd like to stick to using the repository interface. Passing `true` instead of `false` to `SaveBlob` would be enough to force SaveBlob to do actual work. The hashmap lookup in `addPending` shouldn't make much of a difference then.

Reply:
I was under the mistaken impression that that would cause a data structure to grow linearly. Fixed now.